File: SharpMap.Presentation.Web.SharpLayers/Layers/Vector/VectorLayerComponent.cs
using System.Web.UI;
using AjaxControlToolkit;

[assembly: WebResource("SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.js", "text/javascript")
]

namespace SharpMap.Presentation.Web.SharpLayers.Layers.Vector
{
    [
        ClientScriptResource("SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent",
            "SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.js"),
        TargetControlType(typeof (Control))
    ]
    public class VectorLayerComponent : LayerComponent<VectorLayerBuilderParams>
    {
        protected override string ScriptComponentName
        {
            get { return "SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent"; }
        }
    }
}

File: SharpMap.Presentation.Web.SharpLayers/Layers/Vector/VectorLayerComponent.js

Type.registerNamespace('SharpMap.Presentation.Web.SharpLayers.Layers.Vector');

SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent = function() {
    SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.initializeBase(this);


}
SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.prototype = {
    initialize: function() {
        this._builderDelegate = Function.createDelegate(this, this._layerBuilderDelegate);
        SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.callBaseMethod(this, 'initialize');
    },

    dispose: function() {
        SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.callBaseMethod(this, 'dispose');
    },
    _layerBuilderDelegate: function() {
        var options = this.get_builderParams();
        return new OpenLayers.Layer.Vector(this.get_name(), options);
    }

}
SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent.registerClass('SharpMap.Presentation.Web.SharpLayers.Layers.Vector.VectorLayerComponent', SharpMap.Presentation.Web.SharpLayers.Layers.LayerComponent);
